content/tutorials/presence.textile

```diff
@@ -0,0 +1,306 @@
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/ably/tutorials/commits/presence-ruby
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout presence-ruby
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.rb@ and running the demo @bundle exec ruby example.rb@
+
+h2. Next steps
+
+1. If you would like to find out more about how channels, publishing and subscribing works, see the "Realtime channels & messages documentation":/realtime/channels-messages
+1. If you would like to find out more about how presence works, see the "Realtime presence documentation":/realtime/presence
+2. Learn more about "Ably features":https://www.ably.io/features by stepping through our other "Ably tutorials":https://www.ably.io/tutorials
+3. Gain a good technical "overview of how the Ably realtime platform works":https://www.ably.io/documentation/how-ably-works
+4. "Get in touch if you need help":https://www.ably.io/contact
+
+<script src="//cdn.ably.io/lib/ably.min.js"></script>
+<script src="https://apis.google.com/js/api.js"></script>
+<script type="text/javascript">
+  window.addEventListener("load", function() {
+    var ably = new Ably.Realtime({ authUrl: 'https://www.ably.io/ably-auth/token/docs' }),
+        channelName = getQueryParam('channel') || getRandomChannelName(),
+        channel = ably.channels.get(channelName),
+        clientId,
+        $output = $('#output');
+
+    $('#client-id').val(getRandomName());
+
+    ably.connection.on('connecting', function() {
+      log('[Connecting to Ably...]');
+    });
+
+    ably.connection.on('connected', function() {
+      log('[Connected to Ably] ...');
+    });
+
+    channel.attach(function(err) {
+      if(err) {
+        log('[Error attaching to channel: ' + err.toString() + ']');
+      } else {
+        log('[Attached to channel ' + channelName + ']');
+        channel.presence.subscribe(function(msg) {
+          log('[Received presence ' + msg.action + ' from ' + msg.clientId + ', avatar: ' + msg.data + ']');
+          channel.presence.get(function(err, presenceSet) {
+            $('#presence-set').html($.map(presenceSet, function(item) {
+              return '<li><span class="emoji">' + (item.data || '') +'</span> ' + item.clientId + '</li>';
+            }).join());
+          });
+        });
+      }
+    });
+
+    $('button#enter-presence').on('click', function() {
+      clientId = $.trim($('input#client-id').val());
+      var avatar = $('input[name=avatar]:checked').val();
+
+      if(!clientId) {
+        log('You need to specify a client ID before you can enter presence');
+        return;
+      }
+
+      log("[Entering...] ");
+      channel.presence.enterClient(clientId, avatar, function(err) {
```

Comment:
I find it a bit of an add decision to use `enterClient` when that's not really an API anyone would be using, and not the API we've been talking about in the tutorial either.  I would really prefer if we simplified this significantly by breaking the demo into into two stages.

### 1. Enter your name and connect

Name: [pre-populated]
[Connect to Ably]

(section below is disabled or not visible, and once connected, name becomes disabled and connect to ably changes to a disabled button "Connected to Ably..."

### 2. Enter channel "foo" 

Avatar: 👦' 👧' 👨' 👩' [😺 ]
[Enter]

When they press Enter, the button changes to [Leave] with a comment "Choose a new Avatar above to update presence in real time"

If you do this I think all of this code would be simplified immensely.